paddle/fluid/platform/device_context.h

```diff
@@ -39,6 +41,50 @@ limitations under the License. */
 namespace paddle {
 namespace platform {
 
+/*! \brief device temporary allocator singleton */
```

Comment:
document when should temp allocator be used? How is it different from other allocator?